### PR TITLE
Update max-cfi metrics

### PR DESCRIPTION
**Pull request by dfitchett**
## Summary
- Behind feature toggle - NO
- Updates metrics for max ratings (aka max-cfi).
- I'm on the Employee Experience team; we work with the Disability Experience team which maintains Form 526.

## Related issue(s)
- https://github.com/department-of-veterans-affairs/abd-vro/issues/3456

## Testing done
- [x] *New code is covered by unit tests*
- Old behavior: Metrics were selectively published with specific name using the diagnostic code of the condition in the name of the metric.
  - e.g., `vets_api.statsd.api_max_cfi_on_526_started_6260` would be published if a 526 was started with tinnitus (dc 6260) where the segment 'on' is related to the feature flag `disability_526_maximum_rating` is enabled for the user.
- New behavior:
  - the diagnostic code will now be used as a tag for the metrics:
    - `vets_api.statsd.api_max_cfi_526_started` with tag `diagnostic_code:6260`
    - `vets_api.statsd.api_max_cfi_rated_disabilities` with tag `diagnostic_code:6260`
    - `vets_api.statsd.api_max_cfi_submit` with tags `diagnostic_code:6260` and `claimed:true`
- Steps to verify:
  - Once deployed, verify metrics are being published in datadog

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*
NO user impact

## Acceptance criteria
- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog (if applicable)

### CODE REVIEW

File: app/models/concerns/form526_claim_fast_tracking_concern.rb [hunk 1]
```diff
@@ -280,21 +280,22 @@ def update_form_with_classification_code(classification_code)
...
       disability_claimed = diagnostic_codes.include?(diagnostic_code)
-
-      if disability_claimed
-        StatsD.increment("#{MAX_CFI_STATSD_KEY_PREFIX}.#{max_cfi_enabled}.submit.#{diagnostic_code}")
-      end
-      Rails.logger.info('Max CFI form526 submission',
-                        id:, max_cfi_enabled:, disability_claimed:, diagnostic_code:,
-                        cfi_checkbox_was_selected: cfi_checkbox_was_selected?,
-                        total_increase_conditions: increase_disabilities.count)
+      any_claimed ||= disability_claimed
```
Comment (nanotone) [inline app/models/concerns/form526_claim_fast_tracking_concern.rb:286]: I wonder if it'd be more readable to separate it out into something like

```ruby
any_claimed = (max_rated_diagnostic_codes_from_ipf & diagnostic_codes).any?
```

Or even combine it with the `num_max_rated_cfi` value below.

**Review by nanotone — APPROVED** [review]
Just the one suggestion, I'm fine with taking or leaving it.

**Comment by stevenjcumming** [comment]
@dfitchett there are some failing tests

**Comment by dfitchett** [comment]
> @dfitchett there are some failing tests

Turns out the rspec timed out. Most recent run passed.

**Review by stevenjcumming — APPROVED** [review]

**Pull request merged**